test/runtests.jl

```diff
@@ -87,83 +87,72 @@ for i = 2:N
+    results::BlockNLPAlgorithms.BlockSolution,
+)
+    # reset counters
+    initialized_block.cnt = MadNLP.Counters(start_time = time())
+    # solve
+    optimal_solution = MadNLP.optimize!(initialized_block)
+    # overwrite results
+    field_names = fieldnames(typeof(results))
+    for i = 1:length(field_names)
+        setproperty!(results, field_names[i], getproperty(optimal_solution, field_names[i]))
+    end
 end
 
-dual_solution = dual_decomposition(
+dual_solution = solve(
     block_mpc,
+    BlockNLPAlgorithms.DualDecomposition;
     max_iter = 5000,
-    step_size = 0.4,
+    step_size_min = 0.4,
     max_wall_time = 300.0,
     subproblem_solver = MadNLPSolver(print_level = MadNLP.WARN),
     verbosity = 0,
 )
 
-admm_solution = admm(
+admm_solution = solve(
     block_mpc,
+    BlockNLPAlgorithms.ADMM;
     primal_start = zeros(Float64, 2 * N),
-    max_iter = 5000,
-    step_size = 0.4,
+    max_iter = 1000,
+    dynamic_step_size = true,
+    step_size_min = 0.5,
     max_wall_time = 100.0,
     update_scheme = :GAUSS_SEIDEL,
+    verbosity = 1,
     subproblem_solver = MadNLPSolver(print_level = MadNLP.WARN),
 )
 
-@testset "Solve problem using off-the-shelf solvers" begin
-    @test round(admm_solution.objective) ≈ round(dual_solution.objective)
-end
-
-# Implement custom solver
-struct result
-    solution::Vector{Float64}
-    elapsed_time::Float64
-    objective::Float64
-    multipliers::Vector{Float64}
-end
-
-struct MySolver <: AbstractBlockSolver
-    options::Dict{Symbol,<:Any}
-    MySolver(; opts...) = new(Dict(opts))
-end
-
-function BlockNLPAlgorithms.optimize_block!(block::AbstractNLPModel, solver::MySolver)
-    start_time = time()
-    function f!(F, x)
-        F .= grad(block, x)
-    end
 
-    function j!(J, x)
-        J .=
-            sparse(hess_structure(block)[1], hess_structure(block)[2], hess_coord(block, x))
-    end
-
-    sol = nlsolve(f!, j!, zeros(Float64, block.meta.nvar)).zero
-    for i = 1:length(block.meta.nvar)
-        if sol[i] >= block.meta.uvar[i]
-            sol[i] = block.meta.uvar[i]
-        elseif sol[i] <= block.meta.lvar[i]
-            sol[i] = block.meta.lvar[i]
-        end
-    end
-    return result(
-        sol,
-        time() - start_time,
-        obj(block, sol),
-        zeros(Float64, block.meta.ncon),
-    )
-end
-
-my_solution = admm(
+prox_admm_solution = solve(
     block_mpc,
+    BlockNLPAlgorithms.ProxADMM;
     primal_start = zeros(Float64, 2 * N),
     max_iter = 5000,
-    step_size = 0.4,
+    step_size_min = 0.5,
     max_wall_time = 50.0,
-    update_scheme = :GAUSS_SEIDEL,
-    subproblem_solver = MySolver(),
+    update_scheme = :JACOBI,
+    subproblem_solver = MadNLPSolver(print_level = MadNLP.WARN),
 )
 
-@testset "Solve problem using my solver" begin
-    @test ceil(my_solution.objective) ≈ round(admm_solution.objective)
+@testset "Check solver accuracy" begin
+    @test ceil(prox_admm_solution.objective) ≈
```

Comment:
Why do we have to use `ceil` and `round` here?

Reply:
There are slight differences in the objective values produced by the three solution methods.

```julia
  Expression: prox_admm_solution.objective ≈ admm_solution.objective ≈ dual_solution.objective
   Evaluated: 832.9671757447425 ≈ 832.9672929889297 ≈ 832.9672190263699
```